src/gpodder/gtkui/download.py

```diff
@@ -140,31 +192,36 @@ def are_downloads_in_progress(self):
 
         return False
 
+    def move_after(self, iter, position):
+        self.list.move_after(iter, position)
+        iter_task = self.list.get_value(iter, DownloadStatusModel.C_TASK)
+        pos_task = self.list.get_value(position, DownloadStatusModel.C_TASK)
+        self.work_queue.move_after(iter_task, pos_task)
+
+    def move_before(self, iter, position):
+        self.list.move_before(iter, position)
+        iter_task = self.list.get_value(iter, DownloadStatusModel.C_TASK)
+        pos_task = self.list.get_value(position, DownloadStatusModel.C_TASK)
+        self.work_queue.move_before(iter_task, pos_task)
+
     def has_work(self):
-        return any(self._work_gen())
+        return len(self.work_queue) > 0
```

Comment:
reminder: you have implemented `is_empty()` ;-)

Reply:
Good point, I have removed is_empty (it was unused)